apps/src/templates/teacherDashboard/AddSectionDialog.jsx

```diff
@@ -47,6 +64,27 @@ const AddSectionDialog = ({
   const {loginType, participantType} = section || {};
   const title = i18n.newSectionUpdated();
 
+  const onParticipantTypeSelection = participantType => {
+    if (participantType !== 'student') {
+      redirectToNewSectionPage(participantType, SectionLoginType.email);
+    }
+    setParticipantType(participantType);
+  };
+
+  const onLoginTypeSelection = loginType => {
+    // Oauth section types should use the roster dialog, not the section setup page
+    if (
+      [
+        SectionLoginType.picture,
+        SectionLoginType.word,
+        SectionLoginType.email
+      ].includes(loginType)
+    ) {
+      redirectToNewSectionPage(participantType, loginType);
+    }
+    setLoginType(loginType);
```

Comment:
Okay so if oauth, we just keep the current flow and the section is created and they're back on the homepage with a section that's not set up otherwise, right? And then we are keeping the edit section dialog as-is for now? Sorry just trying to keep track of the state we're currently in!

Reply:
Yes for now! I think the plan for launch is to redirect from the roster dialog to the edit page but we don't have an edit page yet :)